mailet/standard/src/main/java/org/apache/james/transport/mailets/MailAttributesToMimeHeaders.java

```diff
@@ -61,11 +63,12 @@ public void init() throws MessagingException {
     public void service(Mail mail) throws MessagingException {
         MimeMessage message = mail.getMessage();
         for (Entry<String, String> entry : mappings.entrySet()) {
-            String value = (String) mail.getAttribute(entry.getKey());
-            if (value != null) {
-                String headerName = entry.getValue();
-                message.addHeader(headerName, value);
-            }
+            AttributeUtils
+                .getValueAndCastFromMail(mail, AttributeName.of(entry.getKey()), String.class)
+                .ifPresent(Throwing.consumer(value -> {
```

Comment:
I propose to add a `map` stage to explicitly extract transformation from the side effect, thus improving readability here...